tests/integration/kubernetes/gha-run.sh

```diff
@@ -253,8 +254,6 @@ function run_tests() {
 
 	# Enable auto-generated policy for CI images that support policy.
 	#
-	# TODO: enable testing auto-generated policy for other types of hosts too.
-	[ "${KATA_HOST_OS}" = "cbl-mariner" ] && export AUTO_GENERATE_POLICY="yes"
```

Comment:
AUTO_GENERATE_POLICY is already used in a few places. Moving this so it gets set in the yaml by each hosts that wants to autogenerate